### PR TITLE
feat(docs/application): :sparkles: Add Java Records concept

### PR DESCRIPTION
**Pull request by pklaschka**
Closes: #164

I've signed the CLA

### CODE REVIEW

File: docs/application/concepts/170-records.mdx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import { Reference, Image } from '/components';
+
+<!-- Brief Description (1-3 lines) of the concept: -->
+
+Records are immutable data classes used in Java to encapsulate "packets" of
+data.
+
+Records make it much easier to implement these kinds of "wrapper classes"
+without having to define constructors, `toString()`, etc., every single time.
```
Comment (cb0s) [inline docs/application/concepts/170-records.mdx:49]: Imo getters and setters are even more important.

**Review by cb0s — COMMENTED** [review]
I think you should also mention the automatic getters and setters at least once in the description as this is one of the core features.

**Comment by cb0s** [comment]
Sorry setters are obviously not possible due to their immutable design. Getters are important though!

File: docs/application/concepts/170-records.mdx (hunk 1 above)
Comment (pklaschka) [reply docs/application/concepts/170-records.mdx]: RE getters: agreed, I'm on it

**Comment by pklaschka** [comment]
Done. Currently, the changes don't show up for me in the PR view, but when you look at the branch, they're already there. Probably some bug with GitHub. I could imagine that they already show up when you see this.

**Review by fussel178 — APPROVED** [review]

**Pull request merged**